reference/conanfile/attributes.rst

```diff
@@ -320,77 +320,98 @@ The ``get_safe()`` method will return ``None`` if that setting or subsetting doe
-values:
+.. important::
+   
+    All the options with their values are encoded into the package ID, as everything that affects the generated binary.
+    See :ref:`method_configure_config_options` and :ref:`method_package_id` methods for information about removing certain
+    options for some configurations.
 
-.. code-block:: python
 
-    class MyPkg(ConanFile):
-        ...
-        options = {"shared": [True, False],
-                   "option1": ["value1", "value2"],}
+A very common one is the option ``shared`` with allowed values of ``[True, False]`` that many recipes declare and use to configure the
+build system and link static or dynamically the dependencies.
 
 Values for each option can be typed or plain strings, and there is a special value, ``ANY``, for
 options that can take any value.
 
-The attribute ``default_options`` has the purpose of defining the default values for the options
-if the consumer (consuming recipe, project, profile or the user through the command line) does
-not define them. It is worth noticing that **an uninitialized option will get the value** ``None``
-**and it will be a valid value if its contained in the list of valid values**. This attribute
-should be defined as a python dictionary too, although other definitions could be valid for
-legacy reasons.
-
 .. code-block:: python
 
     class MyPkg(ConanFile):
         ...
-        options = {"shared": [True, False],
-                   "option1": ["value1", "value2"],
-                   "option2": "ANY"}
-        default_options = {"shared": True,
-                           "option1": "value1",
-                           "option2": 42}
-
-        def build(self):
-            shared = "-DBUILD_SHARED_LIBS=ON" if self.options.shared else ""
-            cmake = CMake(self)
-            self.run("cmake . %s %s" % (cmake.command_line, shared))
-            ...
+        options = {
+            "shared": [True, False],
+            "option1": ["value1", "value2"],
+            "option2": "ANY",
+        }
+
+Every option in a recipe needs to be assigned a value from the ones declared in the ``options`` attribute. The 
+consumer can define options using different methods: command line, profile or consumer
+recipes; **an uninitialized option will get the value** ``None`` **and it will be a valid value if its contained
+in the list of valid values**. Invalid values will produce an error. See attribute :ref:`conanfile_default_options`
+for a way to declare a default value for options in a recipe.
 
 .. tip::
 
     - You can inspect available package options reading the package recipe, which can be
       done with the command :command:`conan inspect mypkg/0.1@user/channel`.
+
+.. tip::
+
     - Options ``"shared": [True, False]`` and ``"fPIC": [True, False]`` are automatically managed in :ref:`cmake_reference` and
       :ref:`autotools_reference` build helpers.
 
+
+**Define the value of an option**
```

Comment:
I think this section needs to explain alternatives in order:

- First, the recommended one in the general case, ``default_options``. Specially for values of the current package (there is no example for this at the moment in the docs).
- In the configure() method. Use an example of forcing the value of a requirement option, not a self.option
- In the conanfile.txt
- In the command line

Reply:
The list is ordered according to precedence as said in the text.

There are examples for `default_options` in the reference entry for that attribute (for options in the same package) and here, for options declared in required packages.